### PR TITLE
nRF52/buttons

### PR DESCRIPTION
**Pull request by niklasad1**
Hi,
I added the following:
- Support for all pushbuttons including the reset button. 
- Added board specific comments
- Moved a couple of comments and added offset intialization in the [chips/uart.rs](https://github.com/niklasad1/tock/blob/nRF52/buttons/chips/nrf52/src/uart.rs)

Furthermore, the reset button was quite tricky because I had to modify non volatile memory in order to change the register that holds the value for the reset button.  

I added two minimal modules in chips [nvmc.rs](https://github.com/niklasad1/tock/blob/nRF52/buttons/chips/nrf52/src/nvmc.rs) and [uicr.rs](https://github.com/niklasad1/tock/blob/nRF52/buttons/chips/nrf52/src/uicr.rs) for this.

### CODE REVIEW

File: chips/nrf52/src/uart.rs [hunk 1]
```diff
@@ -10,6 +10,8 @@ use peripheral_interrupts;
 use peripheral_registers;
 use pinmux;
 
+// this could potentially to replaced to point directly to
```
Comment (alevy) [inline chips/nrf52/src/uart.rs]: "..._be_ replaced to..."

**Comment by alevy** [comment]
Whoa! You need to configure the reset button?? 🤯

**Comment by niklasad1** [comment]
Yes, unfortunately see the links below for info:

[https://devzone.nordicsemi.com/question/54932/nrf52-reset/](https://devzone.nordicsemi.com/question/54932/nrf52-reset/) 
[http://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.tools%2Fdita%2Ftools%2Fnrf5x_command_line_tools%2Fnrf5x_nrfjprogexe_reference.html](http://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.tools%2Fdita%2Ftools%2Fnrf5x_command_line_tools%2Fnrf5x_nrfjprogexe_reference.html)

At least on my board PSELRESET[0] and PSELRESET[1] were 0xFFFF FFFF.

Well it could also be fixed by using nrfjprog:
```bash
$ nrfjprog --memwr 0x10001200 --val 0x00000015 --family nrf52
$ nrfjprog --memwr 0x10001204 --val 0x00000015 --family nrf52
```
But I figured this solution is better mynewt has a similar approach [https://github.com/apache/mynewt-core/blob/master/hw/mcu/nordic/nrf52xxx/src/system_nrf52.c#L179-L192](https://github.com/apache/mynewt-core/blob/master/hw/mcu/nordic/nrf52xxx/src/system_nrf52.c#L179-L192)

I think that snippet is from the Nordic SoftDevice.

**Comment by alevy** [comment]
Oh yeah, this seems 💯 reasonable, I'm just surprised.

**Pull request merged**